events/log.js

```diff
@@ -1,33 +1,36 @@
 const moment = require("moment");
 const chalk = require("chalk");
 
-const clk = new chalk.constructor({ enabled: true });
-
 /* eslint-disable no-use-before-define */
 exports.run = (client, data, type = "log") => {
-  data = resolveObject(data);
-  switch (type.toLowerCase()) {
-    case "debug":
-      console.log(`${clk.bgMagenta(`[${moment().format("YYYY-MM-DD HH:mm:ss")}]`)} ${data}`);
-      break;
-    case "warn":
-      console.warn(`${clk.black.bgYellow(`[${moment().format("YYYY-MM-DD HH:mm:ss")}]`)} ${data}`);
-      break;
-    case "error":
-      console.error(`${clk.bgRed(`[${moment().format("YYYY-MM-DD HH:mm:ss")}]`)} ${data}`);
-      break;
-    case "log":
-      console.log(`${clk.bgBlue(`[${moment().format("YYYY-MM-DD HH:mm:ss")}]`)} ${data}`);
-      break;
-      // no default
-  }
-};
+  type = type.toLowerCase();
 
-function resolveObject(error) {
-  error = error.stack || error.message || error;
+  data = data.stack || data.message || data;
+  if (typeof data === "object" && typeof data !== "string" && !Array.isArray(data)) data = require("util").inspect(data, { depth: 0, colors: true });
+  if (Array.isArray(data)) data = data.join["\n"];
 
-  if (typeof error === "object" && typeof error !== "string") {
-    return require("util").inspect(error, { depth: 0, colors: true });
+  let timestamp = "";
+  if (!client.config.disableLogTimestamps) {
+    timestamp = moment().format("YYYY-MM-DD HH:mm:ss ");
+    if (!client.config.disableLogColor) {
+      switch (type) {
+        case "debug":
+          timestamp = chalk.bgMagenta(timestamp);
+          break;
+        case "warn":
+          timestamp = chalk.black.bgYellow(timestamp);
+          break;
+        case "error":
+          timestamp = chalk.bgRed(timestamp);
+          break;
+        case "log":
+          timestamp = chalk.bgBlue(timestamp);
+          break;
+          // no default
+      }
+    }
   }
-  return error;
-}
+
+  if (type === "debug") type = "log";
+  console[type](data.split("\n").map(str => str.padStart(timestamp.length + str, timestamp)).join("\n"));
```

Comment:
First Arg of padStart should be a number. Adding a number to a string makes it a string. (drop ` + str` to be correct)... But doing that is pointless, you should be concatenating here with template literals.